src/V3EmitCFunc.h

```diff
@@ -106,8 +104,10 @@ class EmitCLazyDecls final : public VNVisitor {
         m_emitter.puts(suffix);
         m_emitter.ensureNewLine();
     }
-    void declared(AstCFunc* nodep) { nodep->user2SetOnce(); }
-    void reset() { AstNode::user2ClearTree(); }
+    void declared(AstCFunc* nodep) {
+        if (alreadyDeclared(nodep)) return;
```

Comment:
`if` and `return` are entirely redundant, but I would just m_emitted.insert() here